versioned_plugins.rb

```diff
@@ -471,6 +484,23 @@ def write_stack_versions(content, type)
       .gsub("%ECS_VERSION%", @ecs_version)
   end
 
+  PLUGIN_DOCS_LAST_GENERATED_FILE = "plugin_docs_last_generated_time.txt"
+
+  def get_docs_last_generated_file_path
+    "#{logstash_docs_path}/#{PLUGIN_DOCS_LAST_GENERATED_FILE}"
+  end
+
+  # Save doc generated time, next time will be used for fetching plugins from this time
+  # Note that if we base on last commit time, PR merge creates new commit where we lose plugin docs between PR creation and merge
+  def save_doc_generated_time
+    # Create if it doesn't exist
+    File.new(get_docs_last_generated_file_path, "w") unless File.exist?(get_docs_last_generated_file_path)
```

Comment:
This line is not needed. the `w` mode in `File.open(get_docs_last_generated_file_path, "w")` will handle the case the file does not exist.

Reply:
TIL:  File.open is a synonym for [File.new](https://ruby-doc.org/core-2.5.5/File.html#method-c-new).
Thanks for pointing this.

Reply:
Removed.